### PR TITLE
feat(chat-e2e): publish folder tests

### PR DESCRIPTION
**Pull request by irinakartun**
**Description:**

publish folder tests

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2597.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2597.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2597.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2597.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2597.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat-e2e/src/core/dialAdminFixtures.ts [hunk 1]
```diff
@@ -117,6 +143,59 @@ const dialAdminTest = dialTest.extend<{
       adminChat.getPublicationReviewControl();
     await use(adminPublicationReviewControl);
   },
+  adminChatHeader: async ({ adminChat }, use) => {
+    const adminChatHeader = adminChat.getChatHeader();
+    await use(adminChatHeader);
+  },
+  adminChatMessages: async ({ adminChat }, use) => {
+    const adminC = adminChat.getChatMessages();
```
Comment (nartovm) [inline apps/chat-e2e/src/core/dialAdminFixtures.ts]: why not const adminChatMessages?

File: apps/chat-e2e/src/tests/cleanup.ts [hunk 2]
```diff
@@ -42,3 +47,32 @@ dialSharedWithMeTest(
...
+      (p) =>
+        p.name?.trim()?.startsWith(publicationRequestPrefix) ||
+        p.name?.trim().startsWith(unpublishRequestPrefix),
+    )) {
+      const requestDetails =
+        await adminPublicationApiHelper.getPublicationRequestDetails(
+          publicationRequest.url,
+        );
+      //if the request is pending un-publication
+      if (requestDetails.resources.some((r) => r.action === 'DELETE')) {
```
Comment (nartovm) [inline apps/chat-e2e/src/tests/cleanup.ts]: I think we should avoid using double filtering - first by name and then by action. We should consider a request with "publicationRequestPrefix" as having an action of "add" and so on.

```
   for (const publicationRequest of publicationRequests.publications) { // Removed the filter here
      const requestDetails =
        await adminPublicationApiHelper.getPublicationRequestDetails(
          publicationRequest.url,
        );

      if (requestDetails.resources.some((r) => r.action === 'DELETE')) {
        await adminPublicationApiHelper.approveRequest(publicationRequest);
      } else if (requestDetails.resources.some((r) => r.action === 'ADD')) {
        await adminPublicationApiHelper.rejectRequest(publicationRequest);
      }
    }
```

**Review by nartovm — CHANGES_REQUESTED** [review]

**Comment by irinakartun** [comment]
/deploy-review
>- **Deployment failed**: [canceled](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/539150)

File: apps/chat-e2e/src/tests/cleanup.ts (hunk 2 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/cleanup.ts]: fixed

File: apps/chat-e2e/src/core/dialAdminFixtures.ts (hunk 1 above)
Comment (irinakartun) [reply apps/chat-e2e/src/core/dialAdminFixtures.ts]: fixed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2597.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2597.nightly-test.deltixhub.io
>E2E tests status: success

**Review by nartovm — DISMISSED** [review]

**Comment by nartovm** [comment]
/deploy-review


>Environment URL: https://chat-ai-dial-chat-pr-2597.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2597.nightly-test.deltixhub.io
>E2E tests status: success

**Review by nartovm — APPROVED** [review]

**Pull request merged**